src/renderer/components/cluster-manager/cluster-status.tsx

```diff
@@ -32,100 +32,92 @@ interface Dependencies {
-  @observable isReconnecting = false;
+  private readonly authOutputs = observable.map<ClusterId, KubeAuthUpdate[]>();
+  private readonly reconnecting = observable.set<ClusterId>();
 
-  constructor(props: ClusterStatusProps & Dependencies) {
-    super(props);
-    makeObservable(this);
-  }
-
-  get cluster(): Cluster {
-    return this.props.cluster;
-  }
+  private readonly authOutput = computed(() => this.authOutputs.get(this.clusterId) ?? []);
+  private readonly hasErrors = computed(() => this.authOutput.get().some(({ isError }) => isError));
+  private readonly isReconnecting = computed(() => this.reconnecting.has(this.clusterId));
+  private readonly clusterName = computed(() => {
+    const entity = catalogEntityRegistry.getById(this.clusterId);
+    const { cluster } = this.props;
 
-  @computed get entity() {
-    return catalogEntityRegistry.getById(this.cluster.id);
-  }
+    return entity?.getName() ?? cluster.name;
+  });
 
-  @computed get hasErrors(): boolean {
-    return this.authOutput.some(({ isError }) => isError);
+  private get clusterId() {
+    return this.props.cluster.id;
   }
 
   componentDidMount() {
     disposeOnUnmount(this, [
-      ipcRendererOn(`cluster:${this.cluster.id}:connection-update`, (evt, res: KubeAuthUpdate) => {
-        this.authOutput.push(res);
+      ipcRendererOn(`cluster:connection-update`, (evt, clusterId: ClusterId, update: KubeAuthUpdate) => {
+        getOrInsert(this.authOutputs, clusterId, []).push(update);
       }),
     ]);
   }
 
-  componentDidUpdate(prevProps: Readonly<ClusterStatusProps>): void {
-    if (prevProps.cluster.id !== this.props.cluster.id) {
-      this.isReconnecting = false;
-      this.authOutput = [];
-    }
-  }
-
-  reconnect = async () => {
-    this.authOutput = [];
-    this.isReconnecting = true;
+  private reconnect = async () => {
+    this.authOutputs.delete(this.clusterId);
+    this.reconnecting.add(this.clusterId);
 
     try {
-      await requestClusterActivation(this.cluster.id, true);
+      await requestClusterActivation(this.clusterId, true);
     } catch (error) {
-      this.authOutput.push({
+      getOrInsert(this.authOutputs, this.clusterId, []).push({
         message: error.toString(),
         isError: true,
       });
     } finally {
-      this.isReconnecting = false;
+      this.reconnecting.delete(this.clusterId);
     }
   };
 
-  manageProxySettings = () => {
-    this.props.navigateToEntitySettings(this.cluster.id, "proxy");
+  private manageProxySettings = () => {
+    this.props.navigateToEntitySettings(this.clusterId, "proxy");
   };
 
-  renderAuthenticationOutput() {
+  private renderAuthenticationOutput() {
     return (
       <pre>
         {
-          this.authOutput.map(({ message, isError }, index) => (
-            <p key={index} className={cssNames({ error: isError })}>
-              {message.trim()}
-            </p>
-          ))
+          this.authOutput
+            .get()
```

Comment:
Nit/idea/opinion: keeping act of MobX de-referencing and the `observer` at same level of abstraction can lead to simpler code.

Eg. here `.get()` is pretty far from `observer` causing uncertainty in reactive context and its granularity.

Reply:
How is is far? It is in the same component...

Reply:
It's a big component :) There may be some benefits to be had if observer and whatever dereferences would be right next to each other. Eg. Intent of code could be more clear, like "ah, so this mutation causes this specific area in UI to update", and not "ah, so this entire component re-renders for a large amount of reasons that are unknown to me".

We've seen this as worthwhile somewhere else, could be here too :)